intel_pytorch_extension_py/launch.py

```diff
@@ -212,22 +214,39 @@ def add_lib_preload(lib_type=None):
                      "/usr/local/lib64/", "/usr/lib/", "/usr/lib64/"]
     lib_find = False
     for lib_path in library_paths:
-        library_file = lib_path + "lib" + lib_type + ".so"
-        matches = glob.glob(library_file)
-        if len(matches) > 0:
-            if "LD_PRELOAD" in os.environ:
-                os.environ["LD_PRELOAD"] = matches[0] + ":" + os.environ["LD_PRELOAD"]
-            else:
-                os.environ["LD_PRELOAD"] = matches[0]
-            lib_find = True
-            break
+        for each_lib_type in lib_type.split(","):
+            library_file = lib_path + "lib" + each_lib_type + ".so"
+            matches = glob.glob(library_file)
+            if len(matches) > 0:
+                if "LD_PRELOAD" in os.environ:
+                    os.environ["LD_PRELOAD"] = matches[0] + ":" + os.environ["LD_PRELOAD"]
+                else:
+                    os.environ["LD_PRELOAD"] = matches[0]
+                lib_find = True
+                break
 
     if not lib_find:
         # Unable to find the TCMalloc library file
         print("Warning: Unable to find the {} library file lib{}.so in $CONDA_PREFIX/lib or  /.local/lib/" 
                " or /usr/local/lib/ or /usr/local/lib64/ or /usr/lib or /usr/lib64 or "
                "~/.local/lib/ so the LD_PRELOAD environment variable will not be set."
                .format(lib_type, lib_type, expanduser("~")))
+
+def get_thp():
+    if os.path.exists(THP_LOC):
+        with open(THP_LOC) as f:
+           val = f.read().strip().split("[")[1].split("]")[0]
+           if val in ['always','never','madvise']:
+               return val
+    return None
+
+def set_thp(arg):
+    if not arg: return
+    if os.path.exists(THP_LOC):
+        os.system("sync;echo 3 > /proc/sys/vm/drop_caches")
+        os.system(f"echo {arg} > {THP_LOC}")
```

Comment:
is it general to all Linux platform about the setting for transparent hugepage ?

Reply:
this is a kernel feature, so it usually common for all linux distros.

Reply:
ok. thx